testutils/src/main/scala/za/co/absa/enceladus/testutils/HelperFunctions.scala

```diff
@@ -133,12 +132,54 @@ object HelperFunctions {
     * Calculates the time it took for passed block of code to execute and finish.
     * @param processToRun Block of code, to be executed
     * @tparam A A type parameter specifying output from processToRun
-    * @return Returns a millisecond difference between start and end time
+    * @return Returns a tuple of milliseconds difference between start and end time
+    *         and return value
     */
-  def calculateTime[A](processToRun: => A): ProcessMeasurement[A] = {
+  def calculateTime[A](processToRun: => A): (Long, A) = {
     val startTime = System.nanoTime()
     val returnValue = processToRun
     val endTime = System.nanoTime()
-    ProcessMeasurement(endTime - startTime, returnValue)
+    val millisecondsPassed = (endTime - startTime) / 1000000
+    (millisecondsPassed, returnValue)
+  }
+
+  /**
+    * Pretty prints elapsed time. If given 91441000 will return "1 day, 1 hour, 24 minutes and 1.00 second"
+    * @param elapsedTime Elapsed time in milliseconds you want to pretty print
+    * @return Returns a string format with human readable time segments
+    */
+  def prettyPrintElapsedTime(elapsedTime: Long): String = {
+    def stringify(count: Double, noun: String, formatter: String = "%.0f"): Option[String] = {
+      if (count == 0) { None }
+      else {
+        val formatted = formatter.format(count)
+        Some(s"$formatted $noun${if (count == 1) { "" } else { "s" }}")
+      }
+    }
+
+    def extractTimeSegment(remaining: Long, multiplier: Long): (Long, Long) = {
+      val count = remaining / multiplier
+      val newRemaining = remaining - (count * multiplier)
+      (count, newRemaining)
+    }
+
+    val minuteMultiplier: Int = 1000 * 60
+    val hourMultiplier: Int = minuteMultiplier * 60
+    val dayMultiplier: Int = hourMultiplier * 24
```

Comment:
Why explicit `Int`? I'm not suggesting to remove it, but just wondering what is the rationale.

Reply:
A remnant of the previous debugging. Will be delted.